rfcs/replace_ssh_with_ssm_for_linux_template/replace_ssh_with_ssm_for_linux_template.md

```diff
@@ -0,0 +1,268 @@
+
+
+### Get Detail Information to Connect
+After user clicking "Connect" blue button, client side will send ``/api/workspaces/service-catalog/${envId}/connections/${connectionId}/ssm-url`` request to server side to get session manager url with siginToken. This request is just for SSM, the server side will have new logic to handle this request. The process of this new logic will have implementation as below:
+* Assume role to get temporary AK/SK
+* Get temporary signin token
+* Generate session manager url with signin token
+
+1. Assume role:
+The ``getCredentialsForRole()`` method is existing in current source code in addons/addon-base/packages/services/lib/aws/aws-service.js file. We can reuse it to get ``EC2SSMRole``'s AK/SK, it will be used for getting signIn token.
+
+
+2. Get temporary signin token
+We can get signIn token through the AK/SK, the code snap is like below:
+```
+  const session = {'sessionId': accessKeyId, 'sessionKey': secretAccessKey 'sessionToken': sessionToken};
+
+  let sessionDuration = 3600; //default session duration, 1 hour
+  if (Number(SessionDuration) > 0) {
+    sessionDuration = Number(SessionDuration);
+  }
+  const getSigninTokenParams={
+    'Action': 'getSigninToken',
+    'SessionDuration': sessionDuration,
+    'Session': session};     
+
+    const getSigninTokenURL = `https://${region}.signin.${consoleDomainSuffix}/federation`;
+
+    const response = await axios.get(getSigninTokenURL, {getSigninTokenParams});
+
+    const signinToken = response.data.SigninToken;
+```  
+
+The sessionDuration can be configured in workspace template input parameter and send to service workbench server side through cloudformation stack output.
+The template should be like below:
+
+```
+Parameters:
+  ...
+  SessionDuration:
+    Type: String
+    Description: Duration of ec2 connection session
+  ...
+```
+
+```
+Outputs:
+  ...
+  SessionDuration:
+    Description: Duration of ec2 connection session
+    Value: !Ref EC2Instance
+```
+
+3. Generate Session Manager URL
+After getting signIn token, we will create session manager url with signIn token and return to client side. User can use this url to access session manager directly.
+Code snap is like below:
+```
+const ssmConsoleUrl = `https://${region}.signin.${consoleDomainSuffix}/federation?Action=login&Issuer=Instruqt&Destination=https://${region}.console.${consoleDomainSuffix}/systems-manager/session-manager/${instanceId}?region=${region}&SigninToken=${signinToken}`;
+
+```
+
+### Redirect to Session Manager page
+Current code already implement the logic of redirecting to the new page with URL from Service Workbench server side. So we can reuse it, and do not need any change.
+
+
+# Summary
+With this refactoring, user can access linux template through Session Manager, does not need to store and maintain private key. And make Service Workbench linux workspace workable in all regions which do not support ec2-instance-connect feature.
+
+# Remaining Work
```

Comment:
remove this sector, use [README of RFC](https://github.com/awslabs/service-workbench-on-aws-cn/blob/develop/rfcs/README.md) for progress tracking

Reply:
Yes, also changed README of RFC content